site/en/docs/extensions/mv3/architecture-overview/index.md

```diff
@@ -48,252 +54,353 @@ as the most important files and the capabilities the extension might use.
+#### Callbacks {: #callbacks }
 
 Most Chrome API methods are asynchronous: they return immediately without waiting for the operation
 to finish. If an extension needs to know the outcome of an asynchronous operation it can pass a
 callback function into the method. The callback is executed later, potentially much later, after the
 method returns.
 
+A method is asynchronous when the callback parameter is available in its signature.
+
+```js
+// Signature for an asynchronous method
+chrome.tabs.query(object queryInfo, function callback)
+```
+
 If the extension needed to navigate the user's currently selected tab to a new URL, it would need to
 get the current tab's ID and then update that tab's address to the new URL.
 
-If the [tabs.query][30] method were synchronous, it may look something like below.
+If the [tabs.query][api-tabs-query] method were synchronous, it may look something like below.
 
+{% Compare 'worse' %}
 ```js
-//THIS CODE DOESN'T WORK
-var tab = chrome.tabs.query({'active': true}); //WRONG!!!
+let tab = chrome.tabs.query(queryOptions); //WRONG!!!
 chrome.tabs.update(tab.id, {url:newUrl});
 someOtherFunction();
 ```
+{% CompareCaption %}
 
-This approach will fail because `query()` is asynchronous. It returns without waiting for the work
-to complete, and does not return a value. A method is asynchronous when the callback parameter is
-available in its signature.
+This approach will fail because `query()` is **asynchronous**. It returns without waiting for the
+work to complete, and does not return a value.
 
-```js
-// Signature for an asynchronous method
-chrome.tabs.query(object queryInfo, function callback)
-```
+{% endCompareCaption %}
+
+{% endCompare %}
 
 To correctly query a tab and update its URL the extension must use the callback parameter.
 
+{% Compare 'better' %}
 ```js
-//THIS CODE WORKS
-chrome.tabs.query({'active': true}, function(tabs) {
+chrome.tabs.query(queryOptions, function(tabs) {
   chrome.tabs.update(tabs[0].id, {url: newUrl});
 });
 someOtherFunction();
 ```
 
+{% endCompare %}
+
 In the above code, the lines are executed in the following order: 1, 4, 2. The callback function
 specified to `query()` is called and then executes line 2, but only after information about the
 currently selected tab is available. This happens sometime after `query()` returns. Although
 `update()` is asynchronous the code doesn't use a callback parameter, since the extension doesn't do
 anything with the results of the update.
 
+#### Promises {: #async }
+
+With the introduction of Manifest V3, many extension API methods now return promises. Not all
+methods in extensions APIs support promises. You can verify whether a method supports promises by
+checking its API reference page.
+
 ```js
-// Synchronous methods have no callback option and returns a type of string
-string chrome.runtime.getURL()
+// Promise
+chrome.tabs.query(queryOptions).then((tabs) => {
+  chrome.tabs.update(tabs[0].id, {url: newUrl});
```

Comment:
```suggestion
chrome.tabs.query(queryOptions)
.then((tabs) => {
  chrome.tabs.update(tabs[0].id, {url: newUrl});
```

Reply:
We prefer to put `then()` on the next line in dcc code samples. I think it makes it easier to see what's going on. It's also less likely to result in an example scrolls horizontally.